src/category_theory/discrete_category.lean

```diff
@@ -14,9 +14,16 @@ def discrete (α : Type u₁) := α
 
 instance discrete_category (α : Type u₁) : small_category (discrete α) :=
 { hom  := λ X Y, ulift (plift (X = Y)),
-  id   := by tidy,
+  id   := λ X, ulift.up (plift.up rfl),
```

Comment:
Why did you replace `tidy` here? I don't really mind, but it seems we should either use `tidy` for both `id` and `comp`, or for neither.

Reply:
I guess it is because `id` is data... even if it is subsingleton...

Reply:
When `tidy` has a morphism `f : X ⟶ Y` in a discrete category, it will do cases on this multiple times, until it finally reduces it to a bunch of constructors to get a morphism of type `X ⟶ X`. I added a rewrite lemma (`id_def`) that rewrites this to the identity morphism `𝟙 X`. It is especially nice if this rewrite rule is by `rfl`, because then `dsimp` can rewrite with it. The proof `tidy` gave was not definitionally equal.

I could give `comp` explicitly, but it's not necessary: after rewriting all morphisms to `𝟙 X`, tidy can rewrite `𝟙 X ≫ 𝟙 X` to `𝟙 X`.

Reply:
I added an explicit definition of `comp`.